classes/task/add_meeting_registrant.php

```diff
@@ -43,9 +43,9 @@ public function execute()
                  JOIN mdl_enrol e ON e.id = ue.enrolid
                  JOIN mdl_role_assignments ra ON ra.userid = u.id
                  JOIN mdl_context ct ON ct.id = ra.contextid AND ct.contextlevel = 50
-                 JOIN mdl_course c ON c.id = ct.instanceid 
+                 JOIN mdl_course c ON c.id = ct.instanceid and e.courseid = c.id
                  JOIN mdl_role r ON r.id = ra.roleid AND r.shortname = 'student'
-                 JOIN mdl_zoom mz ON mz.program_id = c.id AND mz.registration_type = 2
+                 JOIN mdl_zoom mz ON mz.course = c.id AND mz.registration_flag = 1
```

Comment:
@Ash-5 would it make sense to move this flag=1 condition to the main where clause?

Reply:
@meyyappanv resolved.